aata.py

```diff
@@ -39,136 +39,162 @@
 			MakeLinkedArtOrganization
 from settings import *
 
-# Set up environment
-def get_services(**kwargs):
-	return {
-		'trace_counter': itertools.count(),
-		'gpi': create_engine(gpi_engine),
-		'aat': create_engine(aat_engine),
-		'uuid_cache': create_engine(uuid_cache_engine),
-		'fs.data.aata': bonobo.open_fs(aata_data_path)
-	}
-
 ### Pipeline
 
-if DEBUG:
-	sys.stderr.write("In DEBUGGING mode\n")
-	LIMIT		= int(os.environ.get('GETTY_PIPELINE_LIMIT', 10))
-	PACK_SIZE	= 10
-	SRLZ		= Serializer(compact=False)
-	WRITER		= FileWriter(directory=output_file_path)
-	# WRITER	= ArchesWriter()
-else:
-	LIMIT		= 10000000
-	PACK_SIZE	= 10000000
-	SRLZ		= Serializer(compact=True)
-	WRITER		= FileWriter(directory=output_file_path)
-	# WRITER	= ArchesWriter()
-
-
 class AddDataDependentArchesModel(Configurable):
 	models = Option()
 	def __call__(self, data):
 		data['_ARCHES_MODEL'] = self.models['LinguisticObject']
 		return data
 
-def add_articles_chain(graph, records):
-	articles = graph.add_chain(
-		Limit(LIMIT),
-		make_aata_article_dict,
-		add_uuid,
-		add_aata_object_type,
-		MakeLinkedArtLinguisticObject(),
-		AddDataDependentArchesModel(models=arches_models),
-		_input=records.output
-	)
-	if True:
-		# write ARTICLES data
-		graph.add_chain(
-			SRLZ,
-			WRITER,
+class AATAPipeline:
```

Comment:
In the future, this should be abstracted again to just Pipeline to manage cross-project functionality.  Which is not a blocker to merging now, of course.

Reply:
Yeah. The changes I've got pending on top of this PR refactor this a bit, but definitely room for improvement.